mantidproject/mantid
# Create workflow algorithm for polaris pdf calculate

**Description of work.**
This PR creates two workflow algoriths to be used in `Polaris.create_total_scattering_pdf`
`MergeWorkspacesWithLimits` produces the weighted mean of workspaces in a workspace group using two lists of x limits: `XMin` and `XMax`
`CalculateSelfScatteringCorrection` calculates the placzek self scattering correction from a raw polaris workspace, `SampleGeometry`, and `SampleMaterial`, then returns the corrections foccused and nomalized by pixel number using `CalFileName` the path for the correct cal file.

**To test:**

To script below will run the relivant methods in Polaris after being run the Function should output two things: a workspace contrianing the merged corrected spectra and a workspace containing the PDF. Configuration files are needed to run this script, message me on slack and I'll send them over.

```
from mantid.simpleapi import *

import numpy as np

from isis_powder import polaris, SampleDetails
config_file_path = r"C:/Users/wey38795/Documents/polaris-calculate-pdf/polaris_config_example.yaml"
polaris = polaris.Polaris(config_file=config_file_path, user_name="test", mode="PDF")  

sample_details = SampleDetails(height=4.0, radius=0.2985, center=[0, 0, 0], shape='cylinder')
sample_details.set_material(chemical_formula='Si')
polaris.set_sample_details(sample=sample_details)

polaris.create_vanadium(first_cycle_run_no="98532",
                        multiple_scattering=False)
polaris.focus(run_number="98533", input_mode='Summed')

q_lim = np.array([2.5, 3, 4, 6, 7, 3.5, 5, 7, 11, 40]).reshape((2, 5))
polaris.create_total_scattering_pdf(run_number="98533", q_lims=q_lim, merge_banks=True)

```

Fixes #27445
To be merged after #27460
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
